concord_java/src/main/java/com/concord/ComputationWrapper.java

```diff
@@ -0,0 +1,140 @@
+package com.concord;
+
+import com.concord.Metadata;
+import com.concord.ComputationContextImpl;
+import com.concord.StreamTuple;
+import com.concord.swift.*;
+import com.facebook.swift.service.*;
+import com.facebook.nifty.client.FramedClientConnector;
+import com.google.common.net.HostAndPort;
+import org.apache.thrift.TException;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Verify;
+import java.util.*;
+
+public class ComputationWrapper implements ComputationService {
+
+  /** Reference to user defined computation */
+  private final Computation userService;
+
+  /** Thrift client that will perform async RPC */
+  private final BoltProxyService client;
+
+  /** ip and port of proxy */
+  private final Endpoint proxyInfo;
+
+  public ComputationWrapper(final Computation c, final Endpoint proxyInfo) {
+    Preconditions.checkNotNull(c);
+    Preconditions.checkNotNull(proxyInfo);
+    this.userService = c;
+    this.proxyInfo = proxyInfo;
+
+    BoltProxyService client = null;
+    try {
+      FramedClientConnector connector = new FramedClientConnector(
+          HostAndPort.fromParts(proxyInfo.getIp(), proxyInfo.getPort()));
+      client = new ThriftClientManager()
+                   .createClient(connector, BoltProxyService.class)
+                   .get();
+    } catch (Throwable t) {
+      System.err.println("Exception building thrift service: " + t);
+      System.exit(1);
+    }
+    Preconditions.checkNotNull(client);
+    this.client = client;
+    // launch and sleep in thread
+    final ComputationMetadata meta = this.boltMetadata();
+    final BoltProxyService cli = this.client;
+    (new Thread() {
+      public void run() {
+        try {
+          // Need to delay the registration with the scheduler
+          // because the JVM takes too long to spin up a thrift service
+          Thread.sleep(1000);
```

Comment:
first instinct would be to extend our api to have a "handshake" method that we try to call until it succeeds